src/useForm.tsx

```diff
@@ -74,19 +74,28 @@ function createFields<T>(
           value,
           error,
           touched,
+          // For onChange / onBlur, we use a "functional" setState call
+          // (https://reactjs.org/docs/hooks-reference.html#functional-updates)
+          // This avoids issues related to stale state when multiple onChange functions are
+          // invoked within the same render cycle -- e.g.
+          // const onClick = () => { date.day.onChange(...); date.day.onChange(...) }:
           onChange: (updatedValue: any) => {
-            const updatedState = produce(state, updatedState => {
-              set(updatedState, [...path, 'value'], updatedValue)
-              set(updatedState, [...path, 'touched'], true)
-            })
-            setState(updatedState)
+            setState(currentState =>
```

Comment:
So the bug was that our `state` variable would become stale if we tried to change it multiple times within the same tick in the event loop. But React allows us to pass a function to setState, which gives us the up-to-date  value